presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,494 @@
+                    "ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()),
+                    quoted(jdbcColumn.getColumnName()),
+                    quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            // clickhouse versions earlier than 8 do not support the above RENAME COLUMN syntax
+            if (ClickHouseErrorCode.UNKNOWN_EXCEPTION.equals(e.getSQLState())) {
+                throw new PrestoException(NOT_SUPPORTED, format("Rename column not supported in catalog: '%s'", handle.getCatalogName()), e);
+            }
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        return new ClickHouseQueryBuilder(identifierQuote).buildSql(
+                this,
+                session,
+                connection,
+                table.getCatalogName(),
+                table.getSchemaName(),
+                table.getTableName(),
+                columns,
+                table.getConstraint(),
+                split.getAdditionalPredicate(),
+                tryApplyLimit(table.getLimit()));
+    }
+
+    @Override
+    protected void copyTableSchema(Connection connection, String catalogName, String schemaName, String tableName, String newTableName, List<String> columnNames)
+    {
+        String sql = format(
+                "CREATE TABLE %s  ENGINE = Log AS SELECT %s FROM %s WHERE 0 = 1 ",
+                quoted(null, schemaName, newTableName),
+                columnNames.stream()
+                        .map(this::quoted)
+                        .collect(joining(", ")),
+                quoted(null, schemaName, tableName));
+        execute(connection, sql);
+    }
+
+    @Override
+    public void renameTable(JdbcIdentity identity, JdbcTableHandle handle, SchemaTableName newTableName)
+    {
+        // clickhouse doesn't support specifying the catalog name in a rename. By setting the
+        // catalogName parameter to null, it will be omitted in the ALTER TABLE statement.
+        verify(handle.getSchemaName() == null);
+        renameTable(identity, null, handle.getCatalogName(), handle.getTableName(), newTableName);
+    }
+
+    @Override
+    public void dropTable(JdbcIdentity identity, JdbcTableHandle handle)
+    {
+        String sql = "DROP TABLE " + quoted(null, handle.getSchemaName(), handle.getTableName());
+
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public String buildInsertSql(JdbcOutputTableHandle handle)
```

Comment:
We can reuse them from `BaseJdbcClient` right ?

Reply:
ok, I'm trying my best to reuse `BaseJdbcClient` class